src/services/serverApi.ts

```diff
@@ -90,4 +93,132 @@ export const useGetUsers = () => {
   }
 }
 
+//this is used for getting tags for membersSkillUpdateModal
+export const tagsApi = serverApi.injectEndpoints({
+  endpoints: (builder) => ({
+    //Queries
+    getTags: builder.query({
+      query: () => BASE_URL + '/tags',
+    }),
+    getLevels: builder.query({
+      query: () => BASE_URL + '/levels',
+    }),
+    getSkills: builder.query({
+      query: (username) => `${BASE_URL}/users/${username}/skills`,
+      providesTags: ['Skill'],
+    }),
+    //Mutations
+    addNewSkill: builder.mutation({
+      query: (payload) => ({
+        url: '/items',
+        method: 'POST',
+        body: payload,
+      }),
+      invalidatesTags: ['Skill'],
+    }),
+    removeSkills: builder.mutation({
+      query: (payload) => ({
+        url: '/items',
+        method: 'DELETE',
+        body: payload,
+      }),
+      invalidatesTags: ['Skill'],
+      // async onQueryStarted({ tagId, itemId}, {dispatch, queryFulfilled}) {
+      //   const removeResult = dispatch(
+      //     tagsApi.util.updateQueryData('getSkills', itemId, (draft) => {
+      //       return draft?.skills?.filter((skill) => skill.id !== tagId);
+      //     })
+      //   )
+      // },
+    }),
+  }),
+});
+
+export const {
+  useGetTagsQuery,
+  useGetLevelsQuery,
+  useGetSkillsQuery,
+  useAddNewSkillMutation,
+  useRemoveSkillsMutation,
+} = tagsApi;
+
+export const useGetLevels = () => {
+  const { data: tagsData, isLoading: isTagsLoading } = tagsApi.useGetTagsQuery(null);
+  const { data: levelsData, isLoading: isLevelsLoading } = tagsApi.useGetLevelsQuery(null);
+  const tags: tagsType[] = tagsData?.tags;
+  const levels: levelsType[] = levelsData?.levels;
+
+  let tagsWithLevel: tagsWithLevelType[] = [];
+
+  if (isTagsLoading && isLevelsLoading) {
+    return [];
+  } else {
+    for (let i = 0; i < tags?.length; i++) {
```

Comment:
do we really need 2 for loops here

Reply:
Yes as first loop will give us tag and second loop will add levels in it.
`const tagsWithLevel = [
{ name: 'EMBER level 1'},
{ name: 'EMBER level 2'}
]`